lib/melodica_inventory_web/controllers/item_controller.ex

```diff
@@ -48,17 +48,6 @@ defmodule MelodicaInventoryWeb.ItemController do
   defp add_new_item(item_params) do
     Multi.new()
     |> Multi.insert(:item, Item.changeset(%Item{}, item_params))
-    |> Multi.run(:image, fn state -> upload_image(state, item_params["image"]) end)
-  end
-
-  defp upload_image(%{item: %Item{}}, nil), do: {:error, "You need to upload an image"}
-
-  defp upload_image(%{item: %Item{id: item_id}}, %Plug.Upload{path: filename}) do
-    case Cloudex.upload(filename) do
-      [error: error] ->
-        {:error, inspect(error)}
-      [ok: %Cloudex.UploadedImage{public_id: public_id}] ->
-        {:ok, Repo.insert(%Image{public_id: public_id, item_id: item_id})}
-    end
+    |> Multi.run(:image, fn state -> ImageOperations.upload_images(state, item_params["image"], [required: true]) end)
```

Comment:
I am wondering what to do in this case here, as if the upload of some image fails, but the item insert operation succeeds, we can consider the operation as successful and show only the images that managed to upload successfully. Otherwise the uploaded images and the DB records could get out of sync. May be we can make the `ImageOperations.upload_images` always return `{:ok, []}` with all the successfully uploaded images and if an image fails to upload, the user can retry to upload images?